lib/BC/SleighLifter.cpp

```diff
@@ -484,10 +495,97 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
     return LiftStatus::kLiftedInstruction;
   }
 
+  LiftStatus LiftFloatUnop(llvm::IRBuilder<> &bldr, OpCode opc,
+                           VarnodeData *outvar, VarnodeData input_var) {
+    switch (opc) {
+      case OpCode::CPUI_FLOAT_NEG: {
+        auto negate_inval = this->LiftFloatInParam(bldr, input_var);
+        if (negate_inval.has_value()) {
+          return this->LiftStoreIntoOutParam(
+              bldr, bldr.CreateFNeg(*negate_inval), outvar);
+        }
+        break;
+      }
+      case OpCode::CPUI_FLOAT_ABS: {
+        return this->LiftUnaryOpWithFloatIntrinsic(bldr, llvm::Intrinsic::fabs,
+                                                   outvar, input_var);
+      }
+      case OpCode::CPUI_FLOAT_SQRT: {
+        return this->LiftUnaryOpWithFloatIntrinsic(bldr, llvm::Intrinsic::sqrt,
+                                                   outvar, input_var);
+      }
+      case OpCode::CPUI_FLOAT_CEIL: {
+        return this->LiftUnaryOpWithFloatIntrinsic(bldr, llvm::Intrinsic::ceil,
+                                                   outvar, input_var);
+      }
+      case OpCode::CPUI_FLOAT_FLOOR: {
+        return this->LiftUnaryOpWithFloatIntrinsic(bldr, llvm::Intrinsic::floor,
+                                                   outvar, input_var);
+      }
+      case OpCode::CPUI_FLOAT_ROUND: {
+        return this->LiftUnaryOpWithFloatIntrinsic(bldr, llvm::Intrinsic::round,
+                                                   outvar, input_var);
+      }
+      case OpCode::CPUI_FLOAT_NAN: {
+        auto nan_inval = this->LiftFloatInParam(bldr, input_var);
+        if (nan_inval.has_value()) {
+          // LLVM trunk has an `isnan` intrinsic but to support older versions, I think we need to do this.
```

Comment:
@2over12 Is this intrinsic available in llvm 15?

Reply:
I dont think it's in llvm 14 unless I'm missing something

Reply:
What I mean is that if it's in 15, and if remill i s now on 15, then we can assume it's available and drop support for 14.

Reply:
Remill is on 14

Reply:
Okay, file an issue for updating this when the intrinsic support becomes available.